canvas.go

```diff
@@ -270,6 +272,13 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
 // Marshal renders the SVG to the given io.Writer
 func (c *Canvas) Marshal(w io.Writer) {
 	width, height := c.layout()
+	scale := float32(1)
+	if height > maxHeight {
+		scale = maxHeight / float32(height)
+	}
+	if float32(width)*scale > maxWidth {
```

Comment:
That's deliberate, it will only overwrite the scale for height if the width is still too large (after scaling). This can only make the image smaller.